x/gamm/pool-models/balancer/amm.go

```diff
@@ -271,32 +271,46 @@ func (p *Pool) JoinPool(_ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (
 }
 
 // CalcJoinPoolShares calculates the number of shares created to join pool with the provided amount of `tokenIn`.
-// When a single token is provided as an argument, we simply perform single asset join with the token.
-// If tokenIn provided as an argument isn't a sinlge token, it must contain all the tokens in the pool.
-// For the case of multi-asset join for a pool, we first calculate the maximum amount we can join a pool without swap, then
-// perform single asset join for the remaining coins.
-// CalcJoinPoolShares does not directly alter the state of the pool, but only does the calculation for shares for joining the pool.
-func (p *Pool) CalcJoinPoolShares(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, newLiquidity sdk.Coins, err error) {
+// The input tokens must either be:
+// - a single token
+// - contain exactly the same tokens as the pool contains
+//
+// It returns the number of shares created, the amount of coins actually joined into the pool
+// (in case of not being able to fully join), or an error.
+func (p *Pool) CalcJoinPoolShares(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, tokensJoined sdk.Coins, err error) {
+	// 1) Get pool current liquidity + and token weights
+	// 2) If single token provided, do single asset join and exit.
+	// 3) If multi-asset join, first do as much of a join as we can with no swaps.
```

Comment:
Are we mention about when swapFee is used during multi-asset join?

Reply:
added a one line note after the 5 steps.